src/api.cpp

```diff
@@ -89,26 +91,7 @@ struct api_query_string : public std::string {
 // Platform API
 cl_int clGetPlatformIDs(cl_uint num_entries, cl_platform_id* platforms,
                         cl_uint* num_platforms) {
-    LOG_API_CALL("num_entries = %u, platforms = %p, num_platforms = %p",
-                 num_entries, platforms, num_platforms);
-
-    if ((num_platforms == nullptr) && (platforms == nullptr)) {
-        return CL_INVALID_VALUE;
-    }
-
-    if ((num_entries == 0) && (platforms != nullptr)) {
-        return CL_INVALID_VALUE;
-    }
-
-    if (platforms != nullptr) {
-        platforms[0] = gPlatform;
-    }
-
-    if (num_platforms != nullptr) {
-        *num_platforms = 1;
-    }
-
-    return CL_SUCCESS;
+    return clIcdGetPlatformIDsKHR(num_entries, platforms, num_platforms);
```

Comment:
This will result in incorrect logging. I suggest introducing a `cvk_get_platform_ids` function called used by both `clGetPlatformIDs` and `clIcdGetPlatformIDsKHR`. The logging will have to be duplicated.